### PR TITLE
DOC: Reformulate ufunc description in special doc page

### LINKED ISSUE #17078: DOC: "These are not universal functions" difficult to understand for newcomers
### Is your feature request related to a problem? Please describe.

The docs of [special](https://scipy.github.io/devdocs/reference/special.html) include the sentence "These are not universal functions" to point out that a function only accepts scalar arguments and not arrays. I suggest to change the formulation to something a novice numpy user will understand as the concept of "universal function" is a NumPy thing. I guess that even many regular NumPy users do not know the term although they use ufuncs every day.

### Describe the solution you'd like.

Alternative formulation that appeals to newbies and veterans:

"These functions only accept scalar arguments but not arrays. They are not universal functions."

### Describe alternatives you've considered.

_No response_

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by dschmitz89**
#### Reference issue
Closes #17078 

#### What does this implement/fix?
Tried to avoid the usage of "ufunc" as it is an advanced concepts for beginners.

### CODE REVIEW

File: scipy/special/__init__.py [hunk 1]
```diff
@@ -95,7 +100,7 @@
    hankel2e      -- Exponentially scaled Hankel function of the second kind.
    wright_bessel -- Wright's generalized Bessel function.
 
-The following is not a universal function:
+The following function does not accept NumPy arrays:
```
Comment (ev-br) [inline scipy/special/__init__.py:103]: I wonder if there's a way to come up with a wording which both explains it in simple terms, as this PR does, *and* mentions the word ufunc for those readers who are willing to read up or are familiar with numpy to pick up a term.
(to have a cake and eat it too, yes)

Comment (dschmitz89) [reply scipy/special/__init__.py:103]: What about this:

T"he following function does not accept NumPy arrays (it is not a NumPy ufunc)".

Comment (ev-br) [reply scipy/special/__init__.py:103]: This wording works for me and I believe would have worked for me back when I was drowning in various numpy-isms, so a +1 from me. However I'm not necessarily a good proxy and I'm missing the context for this change, so it's my 2 cents.

Comment (dschmitz89) [reply scipy/special/__init__.py]: Updated the wording.

**Review by tupui — APPROVED** [review]
LGTM and Evgeni is also positive about the change. Thanks @dschmitz89

**Pull request merged**